### PR TITLE
created swagger and added pogination in controller

### PR DESCRIPTION
**Pull request by Dima23-ops5**

### CODE REVIEW

File: src/main/java/mate/academy/StringBootIntro/controller/BookController.java [hunk 1]
```diff
@@ -1,49 +1,61 @@
...
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
+    @Operation(summary = "Delete a book", description = "Delete a book by id")
     public void deleteBookById(@PathVariable Long id) {
         bookService.deleteBookById(id);
     }
 
     @GetMapping("/search")
+    @Operation(summary = "Search books", description = "Search books with special parameters")
     public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/controller/BookController.java]:
```suggestion
    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters, @ParameterObject @PageableDefault Pageable pageable) {
```

File: src/main/java/mate/academy/StringBootIntro/controller/BookController.java [hunk 2]
```diff
@@ -1,49 +1,61 @@
...
 @RestController
 @RequestMapping(value = "/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get books", description = "Get all books")
+    public List<BookDto> getAll(Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/controller/BookController.java]:
```suggestion
    public List<BookDto> getAll(@ParameterObject @PageableDefault Pageable pageable) {
```

File: src/main/java/mate/academy/StringBootIntro/controller/BookController.java [hunk 3]
```diff
@@ -1,49 +1,61 @@
...
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get a book", description = "Get a book by id")
     public BookDto getBookById(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/controller/BookController.java]:
```suggestion
    public BookDto getBookById(@PathVariable @Positive Long id) {
```

File: src/main/java/mate/academy/StringBootIntro/controller/BookController.java [hunk 4]
```diff
@@ -1,49 +1,61 @@
...
     @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create a new book", description = "Create a new book")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update a book", description = "Update a book by id")
     public BookDto updateBookByID(@PathVariable Long id,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/controller/BookController.java]:
```suggestion
    public BookDto updateBookByID(@PathVariable @Positive Long id,
```

File: src/main/java/mate/academy/StringBootIntro/controller/BookController.java [hunk 5]
```diff
@@ -1,49 +1,61 @@
...
                                   @RequestBody CreateBookRequestDto createBookRequestDto) {
         return bookService.updateBookById(id, createBookRequestDto);
     }
 
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
+    @Operation(summary = "Delete a book", description = "Delete a book by id")
     public void deleteBookById(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/controller/BookController.java]:
```suggestion
    public void deleteBookById(@PathVariable @Positive Long id) {
```

File: src/main/java/mate/academy/StringBootIntro/controller/BookController.java [hunk 6]
```diff
@@ -1,49 +1,61 @@
...
 import lombok.RequiredArgsConstructor;
 import mate.academy.StringBootIntro.dto.BookDto;
 import mate.academy.StringBootIntro.dto.BookSearchParametersDto;
 import mate.academy.StringBootIntro.dto.CreateBookRequestDto;
 import mate.academy.StringBootIntro.service.BookService;
+import org.springframework.data.domain.Pageable;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
 
+@Tag(name = "Book management", description = "Endpoints for managing books")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/controller/BookController.java:21]:
```suggestion
@Tag(name = "Book management", description = "Endpoints for managing books")
@Validated
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's fix some comments

**Review by sarakhmen — APPROVED** [review]

**Pull request closed without merging**